cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -44,7 +44,7 @@ def provision_vm(provider, template):
 def small_test_vm(setup_provider, provider, small_template, request):
     vm = provision_vm(provider, small_template)
     yield vm
-    vm.cleanup_on_provider()
+    wait_for(lambda: vm.cleanup_on_provider, handle_exception=True, timeout=900)
```

Comment:
Q: can you please explain why we need this at `tear-down`?

Reply:
sure... I encountered a few times an error response from the RHV side that the VM is locked, try again. So that's what I'm doing :)

Reply:
Optional: In such case; I would open this option for all users by taking `wait` as optional argument and handle `wait_for` if user want to wait for proper deletion. 
```
def cleanup_on_provider(wait=False):
   if wait:
        wait_for(---) 
```

Reply:
good idea with one tiny change. wait should accept numbers from 0 to N. if 0 wait shouldnt happen and 0 could be default value